test/extended/util/prometheus/helpers.go

```diff
@@ -45,13 +46,49 @@ type prometheusResponseData struct {
 }
 
 // GetBearerTokenURL makes http request with bearer token
-func GetBearerTokenURL(url, bearer string) (string, error) {
-	cmd := fmt.Sprintf("curl --retry 15 --max-time 2 --retry-delay 1 -s -k -H 'Authorization: Bearer %s' %q", bearer, url)
-	output, err := exec.Command("bash", "-e", "-c", cmd).Output()
+func GetBearerTokenURL(url, bearerToken string) (string, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Timeout:   time.Duration(2 * time.Second),
+		Transport: tr,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+bearerToken)
+
+	const maxRetries = 2
+	var resp *http.Response = nil
+	for retries := 0; retries < maxRetries; retries++ {
```

Comment:
I'd recommend using `wait.PollUntilContextTimeout()` from the `k8s.io/apimachinery/pkg/util/wait` package.

Reply:
The retry loop is replaced by the function`wait.PollUntilContextTimeout()`. Meanwhile the timeout limit is increased to 60 seconds with retry interval of 10 seconds.